core/modules/css/KvInterfaceImplMem.cc

```diff
@@ -92,14 +94,26 @@ KvInterfaceImplMem::KvInterfaceImplMem(string const& filename) {
 KvInterfaceImplMem::~KvInterfaceImplMem() {
 }
 
-void
-KvInterfaceImplMem::create(string const& key, string const& value) {
-    LOGF_DEBUG("create(%1%, %2%)" % key % value);
-    if (exists(key)) {
-        throw KeyExistsError(key);
+std::string
+KvInterfaceImplMem::create(string const& key, string const& value, bool unique) {
+    LOGF_DEBUG("create(%1%, %2%, unique=%3%)" % key % value % int(unique));
+
+    string path = key;
+    if (unique) {
+        // append unique suffix, in-memory KVI is not meant for large-scale
+        // storage, so we can do dumb brute-force loop
+        int seq = 0;
```

Comment:
Couple of reasons not to:
- I do not like random numbers in general (except in simulation)
- we need unique numbers, random is not the same as unique
I'd like to keep that unique number sequential (how it was done in zookeeper) or at least monotonic. It costs one more database query but we should not need to do it very frequently.